### PR TITLE
Factor out onButtonEvent method from controlers

### PR DESCRIPTION
**Pull request by dmarcos**
It factors out the `onButtonEvent` method common to all controls. Other methods can be also factored out like `onButtonChanged` or `onAxisMoved`. I will do follow up PRs with those. I started with `onButtonEvent` for easier review

### CODE REVIEW

File: src/utils/controls.js [hunk 1]
```diff
@@ -0,0 +1,16 @@
+/**
```
Comment (ngokevin) [inline src/utils/controls.js]: There's a tracked-controls utils where all the controllers share code already

File: src/components/vive-controls.js [hunk 2]
```diff
@@ -195,35 +197,18 @@ module.exports.Component = registerComponent('vive-controls', {
...
+    var isTouch;
     if (!this.data.model) { return; }
 
+    isTouch = evtName.indexOf('touch') !== -1;
     // Don't change color for trackpad touch.
     if (isTouch) { return; }
 
     // Update colors.
     color = evtName === 'up' ? this.data.buttonColor : this.data.buttonHighlightColor;
-    if (Array.isArray(buttonName)) {
```
Comment (ngokevin) [inline src/components/vive-controls.js]: Don't need the array handler?

File: src/utils/controls.js (hunk 1 above)
Comment (dmarcos) [reply src/utils/controls.js]: It makes sense for me to separate `tracked-controls` from the higher level controls utils

File: src/components/vive-controls.js (hunk 2 above)
Comment (dmarcos) [reply src/components/vive-controls.js]: I don’t think we need it. I don’t see it used anywhere. I believe we went ahead of ourselves and thought we would neef multiple events when a button is pressed. I can reintroduce but I couldn’t find where this code path is used.

**Comment by dmarcos** [comment]
I think is a bit premature to do a registerController API. There’s plenty of ad-hoc logic to deal with the specifics of each controller and I don’t feel ready for abstraction. Factoring out a few functions will clean up and make code testable. We can reevaluate an`registerController` API a bit later.

File: src/utils/controls.js [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+ * Handle a button event and reemits the events.
+ *
+ * @param {string} id - id of the button.
+ * @param {string} evtName - name of the reemitted event
+ * @param {object} component - reference to the component
+ * @param {string} hand - handedness of the controller: left or right.
+ */
+module.exports.onButtonEvent = function (id, evtName, component, hand) {
```
Comment (ngokevin) [inline src/utils/controls.js]: Concerned about passing in `component` and having a utility call a bunch of the component's methods.

For example, we did this with `checkControllerPresentAndSetup`, I found it hard to follow. The utils depends on `component.el`, `component.controllerPresent`, `component.addEventListeners`, `component.controllerEventsActive`, `component.injectTrackedControls`. There's a huge amount of interface expected to be available.

Since we want to make available these shared methods and have the controllers implement this common interface, we might want to think early about a higher level API/prototype mixin early. Splitting out to a util makes it seem refactored, but the util and component are heavily coupled and not any more testable. It might be the wrong road to go down.

Comment (machenmusik) [reply src/utils/controls.js]: Agree that although the attempt to refactor into utils allowed code reuse, it made it nearly impossible to understand what things were doing without a pretty thorough look

Comment (dmarcos) [reply src/utils/controls.js]: You're correct. It's essentially a mixin but passing the `this` as an argument rather than `this.foo()`. I want to stay focused on the task of improving controls. My goal is to easily be able to add `thumbpadup`, `thumbstickup` methods without having to copy code across controllers. There will be just three methods factored out from controllers. In a separate API we can think of a mixin mechanism that consolidates `checkControllerPresentAndSetup` and these factored out methods in a more sustainable pattern.

Comment (ngokevin) [reply src/utils/controls.js]: I'm not sure the utils improves the code quality of the controls since it makes things hard to understand, I'd favor copy and pasting code. Ideally, if we're refactoring, we might as well refactor in the right direction. We can merge something temporary if we agree on the mechanism and someone promises to do it soon.

Comment (ngokevin) [reply src/utils/controls.js]: Perhaps not agree on the specific code, but that we can have a mixin-style refactor instead soon after you are able to finish the thumbpad/stick work. Then mostly OK with something temporary, these methods are at least less coupled than the tracked-controls utils.

File: src/components/vive-controls.js (hunk 2 above)
Comment (machenmusik) [reply src/components/vive-controls.js]: I believe that was originally there for buttons that were aliased to trigger more than one named event (e.g. 'menu' and 'app') but then we resolved the naming such that there were no duplicates.  So it should not currently be used anywhere

**Comment by dmarcos** [comment]
I consolidated the new methods in `tracked-controls` utils
